src/Symfony/Component/Uid/UuidFactory.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid;
+
+class UuidFactory
+{
+    private $defaultNamedVersion;
+    private $defaultTimedVersion;
+
+    /**
+     * @var Uuid|null
+     */
+    private $defaultNamespace;
+
+    /**
+     * @var Uuid|null
+     */
+    private $defaultNode;
+
+    /**
+     * @param int|null $defaultNamedVersion 5 or 3
+     * @param int|null $defaultTimedVersion 6 or 1
```

Comment:
Should we validate 5/3 and 6/1 ?

Reply:
This is going to be validated by the implementation later on. Let's keep the code clean.